llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -448,6 +448,20 @@ Instruction *InstCombinerImpl::visitMul(BinaryOperator &I) {
   if (match(Op1, m_ZExt(m_Value(X))) && X->getType()->isIntOrIntVectorTy(1))
     return SelectInst::Create(X, Op0, ConstantInt::getNullValue(Ty));
 
+  // mul (sext X), Y -> select X, -Y, 0
+  // mul Y, (sext X) -> select X, -Y, 0
+  Value *SExtOp;
```

Comment:
Can we reuse `X` like the previous matches?

Reply:
My mistake, I completely overlooked the presence of X.